drjt/My-List-Util/t/shuffle.t

```diff
@@ -0,0 +1,23 @@
+use strict;
+use warnings;
+
+use Test::More tests => 14;
+
+BEGIN {
+  require_ok( 'My::List::Util' ) || print "Bail out!\n";
+}
+
+diag( "Testing My::List::Util::shuffle' $My::List::Util::VERSION, Perl $], $^X" );
+ok( defined &My::List::Util::shuffle, 'My::List::Util::shuffle is defined' );
+
+my @in = ( 1..10 );
+my @out = My::List::Util::shuffle( @in );
+# and you shake it all about
+
+is( scalar @in, scalar @out, 'input and output are same length' );
+
+for my $i ( @in ) {
+  ok( $i ~~ @out, "$i is included in \@out" );
+}
+
+isnt( "@in", "@out", 'test input and output are different' ); # intermittent fail - has a 1/3628800 chance of failing
```

Comment:
I'm not enthused by a deliberately coded chance of an intermittent failure, even if it is rare enough that'd you'd likely see intermittent failures for a whole host of other reasons before ever seeing this.